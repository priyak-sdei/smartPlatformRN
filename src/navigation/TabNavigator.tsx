import React from 'react';
import { View } from 'react-native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { Image } from 'react-native';
import { Home, Profile } from '@src/assets/svg';
import HomeScreen from '@src/screens/HomeScreen';
import ProfileScreen from '@src/screens/ProfileScreen';
import { useTheme } from '@shared/theme';

export type TabParamList = {
    Home: undefined;
    Profile: undefined;
};

export type TabImages = {
    Home: any;
    Profile: any;
};

const Tab = createBottomTabNavigator<TabParamList>();

function renderTabIcon(routeName: string, color: string, size: number, tabImages: TabImages) {
    const Icon = tabImages[routeName as keyof TabImages];
    // If it's a function/component (SVG), render as component
    if (typeof Icon === 'function') {
        return <Icon width={size} height={size} fill={color} />;
    }
    // Otherwise, treat as image source
    return (
        <Image
            source={Icon}
            style={{ width: size, height: size, tintColor: color }}
            resizeMode="contain"
        />
    );
}
//add custom images here
function getTabImages() {
    return {
        Home: Home,
        Profile: Profile,
    };
}

interface TabNavigatorProps {
    tabImages?: TabImages;
}

const TabNavigator: React.FC<TabNavigatorProps> = ({ tabImages = getTabImages() }) => {
    const { theme } = useTheme();
    return (
        <View testID="tab-navigator-root" style={{ flex: 1 }}>
            <Tab.Navigator
                screenOptions={({ route }) => ({
                    headerShown: false,
                    tabBarStyle: {
                        backgroundColor: theme.colors.background,
                        borderTopColor: theme.colors.border,
                    },
                    tabBarActiveTintColor: theme.colors.tabActive,
                    tabBarInactiveTintColor: theme.colors.tabInactive,
                    tabBarIcon: ({ color, size }) => renderTabIcon(route.name, color, size, tabImages),
                })}
                initialRouteName="Home"
            >
                <Tab.Screen name="Home" component={HomeScreen} />
                <Tab.Screen name="Profile" component={ProfileScreen} />
            </Tab.Navigator>
        </View>
    );
};

export default TabNavigator; 