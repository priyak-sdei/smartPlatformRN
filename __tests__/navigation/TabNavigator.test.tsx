import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import TabNavigator from '../../src/navigation/TabNavigator';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import { store } from '../../src/redux/store';
import renderer, { act } from 'react-test-renderer';

afterEach(cleanup);

describe('TabNavigator', () => {
    it('renders without crashing', async () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <TabNavigator />
                </NavigationContainer>
            </Provider>
        );
        await waitFor(() => {
            expect(getByTestId('tab-navigator-root')).toBeTruthy();
        });
    });

    it('matches snapshot', () => {
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(
                <Provider store={store}>
                    <NavigationContainer>
                        <TabNavigator />
                    </NavigationContainer>
                </Provider>
            );
        });
        expect(tree!.toJSON()).toMatchSnapshot('TabNavigator component');
    });
}); 