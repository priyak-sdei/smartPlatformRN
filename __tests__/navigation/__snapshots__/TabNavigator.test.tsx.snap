// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNavigator matches snapshot: TabNavigator component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
  testID="tab-navigator-root"
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "flexDirection": "column",
        },
      ]
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    />
    <RNSScreenNavigationContainer
      style={
        {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        onGestureCancel={[Function]}
        sheetAllowedDetents={
          [
            1,
          ]
        }
        sheetCornerRadius={-1}
        sheetElevation={24}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetInitialDetent={0}
        sheetLargestUndimmedDetent={-1}
        style={
          [
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "zIndex": 0,
              },
            ],
            {
              "zIndex": undefined,
            },
          ]
        }
      >
        <View
          aria-hidden={false}
          collapsable={false}
          style={
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
                "flex": 1,
              },
              [
                {
                  "flex": 1,
                },
                [
                  undefined,
                  false,
                ],
              ],
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
              testID="home-root"
            >
              <Text
                style={
                  {
                    "fontSize": 24,
                    "fontWeight": "bold",
                  }
                }
              >
                Home Screen
              </Text>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenNavigationContainer>
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="auto"
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "rgb(216, 216, 216)",
          "borderTopColor": "#E5E5EA",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "end": 0,
          "height": 49,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "paddingTop": 0,
          "position": undefined,
          "start": 0,
          "transform": [
            {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        role="tablist"
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 0,
                "overflow": "visible",
              },
              [
                {
                  "flex": 1,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            accessibilityLabel="Home, tab, 1 of 2"
            accessibilityLargeContentTitle="Home"
            accessibilityShowsLargeContentViewer={true}
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": true,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            href="/Home"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 0,
                "cursor": "pointer",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "opacity": 1,
                "padding": 5,
              }
            }
          >
            <View
              style={
                [
                  {
                    "height": 28,
                    "width": 31,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "minWidth": 25,
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source="test-file-stub"
                  style={
                    {
                      "height": 25,
                      "tintColor": "#d92029",
                      "width": 25,
                    }
                  }
                />
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source="test-file-stub"
                  style={
                    {
                      "height": 25,
                      "tintColor": "#0A0F0F",
                      "width": 25,
                    }
                  }
                />
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "color": "rgb(28, 28, 30)",
                  },
                  {
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  [
                    {
                      "backgroundColor": "transparent",
                      "textAlign": "center",
                    },
                    {
                      "color": "#d92029",
                    },
                    [
                      {
                        "fontSize": 10,
                      },
                      {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              Home
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "borderRadius": 0,
                "overflow": "visible",
              },
              [
                {
                  "flex": 1,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            accessibilityLabel="Profile, tab, 2 of 2"
            accessibilityLargeContentTitle="Profile"
            accessibilityShowsLargeContentViewer={true}
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": false,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            href="/Profile"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 0,
                "cursor": "pointer",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "opacity": 1,
                "padding": 5,
              }
            }
          >
            <View
              style={
                [
                  {
                    "height": 28,
                    "width": 31,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "minWidth": 25,
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source="test-file-stub"
                  style={
                    {
                      "height": 25,
                      "tintColor": "#d92029",
                      "width": 25,
                    }
                  }
                />
              </View>
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": "100%",
                    },
                    {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source="test-file-stub"
                  style={
                    {
                      "height": 25,
                      "tintColor": "#0A0F0F",
                      "width": 25,
                    }
                  }
                />
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "color": "rgb(28, 28, 30)",
                  },
                  {
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  [
                    {
                      "backgroundColor": "transparent",
                      "textAlign": "center",
                    },
                    {
                      "color": "#0A0F0F",
                    },
                    [
                      {
                        "fontSize": 10,
                      },
                      {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ],
                ]
              }
            >
              Profile
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
